core/src/main/java/org/elasticsearch/indices/recovery/RecoveryTargetService.java

```diff
@@ -263,12 +263,6 @@ public RecoveryResponse newInstance() {
                 return;
             }
 
-            if (cause instanceof IndexShardClosedException) {
```

Comment:
I'm not sure we should change this as part of this PR (I recall some issues when I tried that, will have to check again). I can leave this dead code for now and add a TODO instead.